pkg/systemd/ssh-list-host-keys.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -eu
+port="22"
+host="localhost"
+
+parse_addr() {
+    if [ -n "$1" ]; then
+        count=$(echo "$1" | grep -o ":" | wc -l)
+        if [ "$count" -ne "0" ]; then
+            p_spot=$(($count + 1))
+            p=$(echo "$1" | cut -d ":" "-f$p_spot")
+            h=$(echo "$1" | cut -d ":" "-f-$count")
+        else
+            p=$1
+            h="localhost"
+        fi
+
+        if [ "$p" -eq "$p" ] 2>/dev/null; then
+            port="$p"
+            host="$h"
+            if [ "$host" == "[::]" ]; then
+                host="::"
+            fi
+        fi
+    fi
+}
+
+# Try to find where sshd might be listening
+
+# Check sshd_config, only works for root
+$(sshd -T | grep "listenaddress " | cut -d' ' -f2-) | while IFS='\n' read line; do
```

Comment:
This doesn't catch the "port" setting in sshd_config. In addition since you're using ```while``` we can:

```
sshd -T || true | while read name value; do
     if [ "$name" = "listenstream" -o "$name" = "port" ]; then
         parse_addr "$value"
     fi
done
```